components/channel_notifications_modal/channel_notifications_modal.tsx

```diff
@@ -14,47 +13,43 @@ import {IgnoreChannelMentions, NotificationLevels, NotificationSections} from 'u
 
 import NotificationSection from 'components/channel_notifications_modal/components/notification_section.jsx';
 
-export default class ChannelNotificationsModal extends React.PureComponent {
-    static propTypes = {
-
-        /**
-         * Function that is called when the modal has been hidden and should be removed
-         */
-        onExited: PropTypes.func.isRequired,
-
-        /**
-         * Object with info about current channel
-         */
-        channel: PropTypes.object.isRequired,
-
-        /**
-         * Object with info about current channel membership
-         */
-        channelMember: PropTypes.object.isRequired,
-
-        /**
-         * Object with info about current user
-         */
-        currentUser: PropTypes.object.isRequired,
-
-        /**
-         * Boolean whether server sends push notifications
-         */
-        sendPushNotifications: PropTypes.bool.isRequired,
-
-        /*
-         * Object with redux action creators
-         */
-        actions: PropTypes.shape({
-
-            /*
-             * Action creator to update channel notify props
-             */
-            updateChannelNotifyProps: PropTypes.func.isRequired,
-        }),
-    };
-
-    constructor(props) {
+import {Channel, ChannelNotifyProps} from '@mattermost/types/channels';
+import {UserNotifyProps, UserProfile} from '@mattermost/types/users';
+
+import {PropsFromRedux} from '.';
```

Comment:
```suggestion
import type {PropsFromRedux} from './index';
```
So we dont have cyclic dependency in js